src/Access/LDAPAccessStorage.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <Access/MemoryAccessStorage.h>
+#include <Core/Types.h>
+#include <ext/scope_guard.h>
+#include <mutex>
+#include <set>
+
+
+namespace Poco
+{
+    namespace Util
+    {
+        class AbstractConfiguration;
+    }
+}
+
+
+namespace DB
+{
+class AccessControlManager;
+
+/// Implementation of IAccessStorage which allows attaching users from a remote LDAP server.
+/// Currently, any user name will be treated as a name of an existing remote user,
+/// a user info entity will be created, with LDAP_SERVER authentication type.
+class LDAPAccessStorage : public IAccessStorage
```

Comment:
I've added the function `IAccessStorage::getStorageParamsJSON()` (https://github.com/ClickHouse/ClickHouse/pull/14828) which returns the parameters of an access storage which should be visible in the system table `system.user_directories`. By default it returns `"{}"` but I think you could override it in `LDAPAccessStorage`.

Reply:
Implemented.